flytekit/tools/module_loader.py

```diff
@@ -118,16 +158,10 @@ def iterate_registerable_entities_in_order(
         include_entities = tuple(list(include_entities or set()))
 
     entity_to_module_key = {}
-    for m in iterate_modules(pkgs):
-        for k in dir(m):
-            o = m.__dict__[k]
-            if isinstance(o, _registerable.RegisterableEntity) and not o.has_registered:
-                if o.instantiated_in == m.__name__:
-                    entity_to_module_key[o] = (m, k)
-                    if isinstance(o, _SdkRunnableWorkflow) and o.should_create_default_launch_plan:
-                        # SDK should create a default launch plan for a workflow.  This is a special-case to simplify
-                        # authoring of workflows.
-                        entity_to_module_key[o.create_launch_plan()] = (m, k)
+    if pkgs is not None:
```

Comment:
Should we use pkgs within directory or somehow try to respect this option?  I'm not sure - I've always found this option a bit weird, but I think people do use it.

Running outside of the container, without this, serialize will just run through everything underneath the given directory right?

Reply:
yeah I waffled on that too. I'm not sure either but I really can't fathom the need where someone only wants to upload specific packages in a directory and exclude others

Reply:
And yet people do, they use it all the time for some reason, even Spotify asked about it.  At the very least, let's read in the value of the configuration and write out a very stern WARNING about what's happening.

Reply:
ohh absolutely, when we are trying to iterate limiting to only a few packages is very useful - no? - the ones that I am specifically workin on as a user

Reply:
sure I guess my question is how often can that not be solved by just using a more specific directory rather than multiple packages? let me add the pkgs option though since the consensus seems to be it's useful to hace